packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -136,56 +129,122 @@ export function Comments( {
 				}
-				// The thread's starting top position is determined by its
-				// associated block's position.
-				const blockElement = blockRefs[ thread.id ];
-				const blockRect = blockElement?.getBoundingClientRect();
-				const threadTop = blockRect?.top || 0;
 
-				// Heights are tracked by the comment threads themselves.
+				blockElement = blockRefs[ thread.id ];
+				blockRect = blockElement?.getBoundingClientRect();
+				const threadTop = blockRect?.top || 0;
 				const threadHeight = heights[ thread.id ] || 0;
 
-				// By default, remove the top margin by shifting the block up
-				// so it more precisely aligns with the block.
 				let additionalOffset = -16;
 
-				// The first block never needs to be adjusted.
-				if ( previousThreadData ) {
-					// Check if the thread overlaps with the previous one.
-					const previousBottom =
-						previousThreadData.threadTop +
-						previousThreadData.threadHeight;
-					if ( threadTop < previousBottom ) {
-						// Shift down by the difference plus a margin to avoid overlap.
-						additionalOffset = previousBottom - threadTop + 20;
-					}
+				// Check if the thread overlaps with the previous one.
+				const previousBottom =
+					previousThreadData.threadTop +
+					previousThreadData.threadHeight;
+				if ( threadTop < previousBottom ) {
+					// Shift down by the difference plus a margin to avoid overlap.
+					additionalOffset = previousBottom - threadTop + 20;
 				}
 
-				// Store the current thread's position and height for the next iteration.
+				offsets[ thread.id ] = additionalOffset;
+
+				// Update for next iteration.
 				previousThreadData = {
 					threadTop: threadTop + additionalOffset,
 					threadHeight,
 				};
+			}
+
+			// Process threads before the selected thread, offsetting any overlapping
+			// threads upward.
+			let nextThreadData = {
+				threadTop: selectedThreadTop - 16,
+				threadHeight: selectedThreadHeight,
+			};
+
+			for ( let i = selectedThreadIndex - 1; i >= 0; i-- ) {
+				const thread = threads[ i ];
+				if ( ! blockRefs[ thread.id ] ) {
+					continue;
+				}
+
+				blockElement = blockRefs[ thread.id ];
+				blockRect = blockElement?.getBoundingClientRect();
+				const threadTop = blockRect?.top || 0;
+				const threadHeight = heights[ thread.id ] || 0;
+
+				let additionalOffset = -16;
+
+				// Calculate the bottom position of this thread with default offset.
+				const threadBottom = threadTop + threadHeight;
+
+				// Check if this thread's bottom would overlap with the next thread's top.
+				if ( threadBottom > nextThreadData.threadTop ) {
+					// Shift up by the difference plus a margin to avoid overlap.
+					additionalOffset =
+						nextThreadData.threadTop -
+						threadTop -
+						threadHeight -
+						20;
+				}
 
 				offsets[ thread.id ] = additionalOffset;
-			} );
 
+				// Update for next iteration (going upward).
+				nextThreadData = {
+					threadTop: threadTop + additionalOffset,
+					threadHeight,
```

Comment:
As far as I can tell, `nextThreadData.threadHeight` value isn't used anywhere.

Reply:
I'll double check and remove it.